### PR TITLE
Phase out the client DSL in the docs

### PR DESCRIPTION
**Pull request by gringrape**
Use Request(...) directly instead of using the DSL like Post(...), Get(...)

### CODE REVIEW

**Comment by armanbilge** [comment]
Great, thanks for getting the PR up!!

According to CI there's still a couple issues.
```
error: client.md:375:1: overloaded method expect with alternatives:
  (s: String)(implicit d: org.http4s.EntityDecoder[cats.effect.IO,String]): cats.effect.IO[String] <and>
  (uri: org.http4s.Uri)(implicit d: org.http4s.EntityDecoder[cats.effect.IO,String]): cats.effect.IO[String] <and>
  (req: cats.effect.IO[org.http4s.Request[cats.effect.IO]])(implicit d: org.http4s.EntityDecoder[cats.effect.IO,String]): cats.effect.IO[String] <and>
  (req: org.http4s.Request[cats.effect.IO])(implicit d: org.http4s.EntityDecoder[cats.effect.IO,String]): cats.effect.IO[String]
 cannot be applied to (org.http4s.Request[Nothing])
httpClient.expect[String](request)
^^^^^^^^^^^^^^^^^^^^^^^^^
error: client.md:392:14: type mismatch;
 found   : org.http4s.EntityEncoder[Nothing,org.http4s.UrlForm]
 required: org.http4s.EntityEncoder[F,org.http4s.UrlForm]
Note: Nothing <: F, but trait EntityEncoder is invariant in type F.
You may wish to define F as +F instead. (SLS 4.5)
  .withEntity(
             ^
error: client.md:400:1: overloaded method expect with alternatives:
  (s: String)(implicit d: org.http4s.EntityDecoder[cats.effect.IO,App.this.AuthResponse]): cats.effect.IO[App.this.AuthResponse] <and>
  (uri: org.http4s.Uri)(implicit d: org.http4s.EntityDecoder[cats.effect.IO,App.this.AuthResponse]): cats.effect.IO[App.this.AuthResponse] <and>
  (req: cats.effect.IO[org.http4s.Request[cats.effect.IO]])(implicit d: org.http4s.EntityDecoder[cats.effect.IO,App.this.AuthResponse]): cats.effect.IO[App.this.AuthResponse] <and>
  (req: org.http4s.Request[cats.effect.IO])(implicit d: org.http4s.EntityDecoder[cats.effect.IO,App.this.AuthResponse]): cats.effect.IO[App.this.AuthResponse]
 cannot be applied to (org.http4s.Request[Nothing])
httpClient.expect[AuthResponse](postRequest)
```

**Comment by gringrape** [comment]
@armanbilge 

I failed to build locally because of below issue.

I tried changing the versions of scala, jdk but none works.

Do you have any idea about this issue? Could you give any hint?

OS:
```
macOS Monterey 12.6 Apple silicon
```
Versions:
```
sbt 1.8.2 
Eclipse Adoptium Java 19.0.2
Scala 3.2.0
```


```
java.util.NoSuchElementException: key not found: (0,23)
	at scala.collection.immutable.Map$EmptyMap$.apply(Map.scala:107)
	at scala.collection.immutable.Map$EmptyMap$.apply(Map.scala:105)
	at org.http4s.sbt.Http4sSitePlugin$.$anonfun$projectSettings$5(Http4sSitePlugin.scala:52)
	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
	at sbt.internal.util.EvaluateSettings$MixedNode.evaluate0(INode.scala:228)
	at sbt.internal.util.EvaluateSettings$INode.evaluate(INode.scala:170)
	at sbt.internal.util.EvaluateSettings.$anonfun$submitEvaluate$1(INode.scala:87)
	at sbt.internal.util.EvaluateSettings.sbt$internal$util$EvaluateSettings$$run0(INode.scala:99)
	at sbt.internal.util.EvaluateSettings$$anon$3.run(INode.scala:94)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1589)
```

**Comment by armanbilge** [comment]
> Do you have any idea about this issue? Could you give any hint?

Yes, it's not related to your system. It appears you are missing all the tags from the repository (try running `git tag`).

To restore them you can run a command like `git pull upstream --tags`.

**Comment by gringrape** [comment]
> > Do you have any idea about this issue? Could you give any hint?
> 
> Yes, it's not related to your system. It appears you are missing all the tags from the repository (try running `git tag`).
> 
> To restore them you can run a command like `git pull upstream --tags`.

Thank you for your advice! It works.

**Review by armanbilge — APPROVED** [review]
LGTM, I think this will be a lot clearer for new users. Thanks for your work on this!

**Review by danicheg — APPROVED** [review]
Thanks, this looks good to me!

File: docs/docs/dsl.md [hunk 1]
```diff
@@ -474,7 +473,7 @@ val greetingWithIdService = HttpRoutes.of[IO] {
 
 ```scala mdoc
 greetingWithIdService
-  .orNotFound(GET(uri"/hello/name;first=john;last=doe;id=123/greeting"))
+  .orNotFound(Request[IO](Method.GET, uri"/hello/name;first=john;last=doe;id=123/greeting"))
```
Comment (valencik) [inline docs/docs/dsl.md:476]: This line is also a little too long

**Comment by armanbilge** [comment]
> I've left a couple requests for shorter lines

Project for another day, but it would be good if we could enforce a column width somehow for code blocks. I have this issue with long lines on my own projects as well.

**Review by valencik — APPROVED** [review]
Thanks! Looks great!

**Pull request merged**